tests/presets.spec.js

```diff
@@ -1248,6 +1248,891 @@ describe('Presets', function() {
+              'VeRsIoN = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+      });
+
+      describe('file with a non-semver version', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.versionedFile = path.join(this.cwd.name, 'File.ver');
+
+          fs.writeFileSync(this.versionedFile, [
+            'blah blah',
+            'version = "not-a-semver"',
+            'bloop bloop',
+            '',
+            'foo foo',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.versionedFile);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to update the version', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            regex: /version\s*=\s*/
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+      });
+
+      describe('file in a subdirectory', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.subDir = path.join(this.cwd.name, 'subdirectory');
+          this.versionedFile = path.join(this.subDir, 'File.ver');
+
+          fs.mkdirSync(this.subDir);
+          fs.writeFileSync(this.versionedFile, [
+            'blah blah',
+            'version = "1.0.0"',
+            'bloop bloop',
+            '',
+            'foo foo',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.versionedFile);
+          fs.rmdirSync(this.subDir);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to update the version when specifying the subdirectory', function(done) {
+          presets.updateVersion.quoted({
+            file: 'File.ver',
+            baseDir: 'subdirectory',
+            regex: /version\s*=\s*/
+          }, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.versionedFile, 'utf8')).to.equal([
+              'blah blah',
+              'version = "1.1.0"',
+              'bloop bloop',
+              '',
+              'foo foo',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+        it('should be able to update the version when adding the subdirectory to the filename', function(done) {
+          presets.updateVersion.quoted({
+            file: 'subdirectory/File.ver',
```

Comment:
I guess this should have been `path,join('subdirectory', 'File.ver')` - oh well, doesn't matter :)